OMCompiler/Compiler/SimCode/SimCodeUtil.mo

```diff
@@ -9144,17 +9169,20 @@ algorithm
         isDiscrete = BackendVariable.isVarDiscrete(dlowVar);
         arrayCref = ComponentReference.getArrayCref(cr);
         aliasvar = getAliasVar(dlowVar, optAliasVars);
-        caus = getCausality(dlowVar, vars);
         numArrayElement = List.map(inst_dims, ExpressionDump.dimensionIntString);
         // print("name: " + ComponentReference.printComponentRefStr(cr) + "indx: " + intString(indx) + "\n");
         // check if the variable has changeable value
         // parameter which has final = true or evaluate annotation are not changeable
         isValueChangeable = ((not BackendVariable.hasVarEvaluateTrueAnnotationOrFinalOrProtected(dlowVar)
                             and (BackendVariable.varHasConstantBindExp(dlowVar) or not BackendVariable.varHasBindExp(dlowVar))))
                             and isFixed;
+        caus = getCausality(dlowVar, vars, isValueChangeable);
+        variability = SimCodeVar.FIXED(); // PARAMETERS()
```

Comment:
A top-level input parameter should probably be `TUNABLE()`.

Reply:
TUNABLE()  are is still not clear,  From the specification as far i understand, they are almost similar to parameters, we can do this later, in a new pull request